enzyme/test/Enzyme/ForwardModeVector/sub.ll

```diff
@@ -0,0 +1,38 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -enzyme-preopt=false -mem2reg -instsimplify -instcombine -simplifycfg -S | FileCheck %s
+
+%struct.Gradients = type { double, double }
+
+; Function Attrs: nounwind
+declare %struct.Gradients @__enzyme_fwddiff(double (double, double)*, ...)
+
+; Function Attrs: noinline nounwind readnone uwtable
+define double @tester(double %x, double %y) {
+entry:
+  %0 = fsub fast double %x, %y
+  ret double %0
+}
+
+define %struct.Gradients @test_derivative(double %x, double %y) {
+entry:
+  %0 = tail call %struct.Gradients (double (double, double)*, ...) @__enzyme_fwddiff(double (double, double)* nonnull @tester, metadata !"enzyme_dupn", i64 2, double %x, double 1.0, double 0.0, metadata !"enzyme_dupn", i64 2, double %y, double 0.0, double 1.0)
```

Comment:
Is it useful to have dupn instead of dup? We can have a single var for the width like enzyme_width, and then all dup's are defined to be the corresponding width?

Reply:
Sure let's do this.